lib/Simulation/Simulator.py

```diff
@@ -236,41 +297,35 @@ def summarize(self):
 
             
     def extract(self):
-        current_time = datetime.datetime.now()
-        path = current_time.strftime("%Y%m%d-%H%M")
-        os.mkdir(path)
-        
+        # This function always open and close the files 
+        # to guarantee that they are being rewriten from scratch
 
-        with open(join(path,'infection_summary.csv'), 'w', newline='') as csvfile:
-            fieldnames = ['Day','Hour','Minutes','CurrentStep','Susceptible','Exposed','Infectious','Recovered']
-            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+        summaryFilePath = join(self.reportPath,'infection_summary.csv')
+        with open(summaryFilePath, 'w', newline='') as summaryFile:
+            writer = csv.DictWriter(summaryFile, fieldnames=summaryFieldnames)
             writer.writeheader()
             for x in self.infectionHistory:
                 writer.writerow(x)
-                
-        infectionDetails = []        
-        for i in self.agents:
-            if (i.infection is not None):
-                infectionDetails.append(i.infection.summarize())
-         
-        with open(join(path,'infection_details.csv'), 'w', newline='') as csvfile:
-            fieldnames = ['infectedAgentId','infectedAgentProfession','originAgentId','originAgentProfession','location','lat','lon','nodeId'
-,"exposedTimeStamp","exposedDay","exposedHour","exposedMinutes","incubationDuration"                   
-,"infectiousTimeStamp","infectiousDay","infectiousHour","infectiousMinutes","recoveryDuration"                   
-,"recoveredTimeStamp","recoveredDay","recoveredHour","recoveredMinutes"            
-,"symptomaticTimeStamp","symptomaticDay","symptomaticHour","symptomaticMinutes"            
-,"severeTimeStamp","severeDay","severeHour","severeMinutes"]
-            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+            summaryFile.close()
+        
+        detailsFilePath = join(self.reportPath,'infection_details.csv')
+        with open(detailsFilePath, 'w', newline='') as detailsFile:
+            writer = csv.DictWriter(detailsFile, fieldnames=detailsFieldnames)
             writer.writeheader()
-            for x in infectionDetails:
-                writer.writerow(x)
+            for i in self.agents:
+                if (i.infection is not None):
+                    summary = i.infection.summarize()
+                    writer.writerow(summary)
+            detailsFile.close()
          
-        with open(join(path,'agents.csv'), 'w', newline='') as csvfile:
-            fieldnames = getAgentKeys()
-            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+        agentsFilePath = join(self.reportPath,'agents.csv')
+        with open(agentsFilePath, 'w', newline='') as agentsFile:
+            agentFieldnames = getAgentKeys()
+            writer = csv.DictWriter(agentsFile, fieldnames=agentFieldnames)
             writer.writeheader()
             for x in self.agents:
```

Comment:
TO Issue: Consider if we want to keep the history of the agents in the future (maybe this is too much data)